src/binary/uint.rs

```diff
@@ -73,6 +69,45 @@ impl UInt {
     }
 }
 
+/// The result of the [`magnitude_to_bytes`] function. The fields are private;
+/// users must go through `as_ref()` to see the slice that ignores leading
+/// zeros.
+///
+/// This type exists to avoid heap allocation.
+pub struct U64Magnitude {
+    magnitude_bytes: [u8; mem::size_of::<u64>()],
+    first_occupied_byte: usize,
```

Comment:
Probably not worth optimizing, but `usize` is overkill for sure, `u8` is all that is needed, but even `u32` probably aligns better (`u8` probably doesn't matter because of padding).

Reply:
It has to become a usize at some point because of the `RangeFrom` implementation. As an aside, I didn't change this at all in this PR, I just de-duplicated the existing implementation.